cdap-docs/faq/source/cdap-startup.rst

```diff
@@ -0,0 +1,226 @@
+If you've installed the ``cdap-cli`` RPM or Deb, it's located under ``/opt/cdap/cli/bin``.
+You can add this location to your PATH to prevent the need for specifying the entire script every time.
+
+**Note:** These commands will list the contents of the package ``cdap-cli``, once it has
+been installed::
+
+  rpm -ql cdap-cli
+  dpkg -L cdap-cli
+
+.. _faq-installation-startup-memory-core-requirements:
+
+What are the memory and core requirements for CDAP?
+---------------------------------------------------
+The settings are governed by two sources: CDAP and YARN. The default setting for CDAP are
+found in the ``cdap-defaults.xml``, and are over-ridden in particular instances by the
+``cdap-site.xml`` file. These vary with each service and range from 512 to 1024 MB and
+from one to two cores.
+
+The YARN settings will over-ride these; for instance, the minimum YARN container size is
+determined by ``yarn.scheduler.minimum-allocation-mb``. The YARN default in ``HDP/Hadoop``
+is 1024 MB, so containers will be allocated with 1024 MB, even if the CDAP settings are
+for 512 MB.
+
+Can a current CDAP installation be upgraded more than one version?
+------------------------------------------------------------------
+This table lists the upgrade paths available for different CDAP versions:
+
++---------+---------------------+
+| Version | Upgrade Directly To |
++=========+=====================+
+| 3.1.x   | 3.2.x               |
++---------+---------------------+
+| 3.0.x   | 3.1.x               |
++---------+---------------------+
+| 2.8.x   | 3.0.x               |
++---------+---------------------+
+| 2.6.3   | 2.8.2               |
++---------+---------------------+
+
+If you are doing a new installation, we recommend using the current version of CDAP.
+
+
+Configuring Distributed Mode
+============================
+
+Are at least two machines really required?
+------------------------------------------
+The CDAP components are independently scalable, so you can install from 1 to *N* of each
+component on any combination of nodes.  The primary reasons for using at least two
+machines are for HA (high availability) and for ``cdap-router``'s data ingest capacity.
+
+It is not necessary to install all components on both machines; you could install just the
+CDAP UI on a third machine with other components on the second node. You can install each
+component on a separate machine (or more) if you choose. The :ref:`HA [High Availability]
+Environment diagram <deployment-architectures-ha>` gives just one possible
+configuration.
+
+My Hive Server2 defaults to 10000; what should I do?
+----------------------------------------------------
+By default, CDAP uses port 10000. If port 10000 is being used by another service, simply
+change the ``router.bind.port`` in the ``cdap-site.xml`` to another available port. Since
+in the Hadoop ecosystem, Hive Server2 defaults to 10000, we are considering changing the
+router default port. 
```

Comment:
The Ambari service, at least, will detect this and run Router on 11015... our current recommendation is simply to run CDAP on a different host than HiveServer2 (one word, by the way)... Link to CDAP-1696